api/views.py

```
@@ -90,3 +95,47 @@ def put(self, request):
             status_code = status.HTTP_200_OK if brand_instance else status.HTTP_201_CREATED
             return Response(serializer.data, status=status_code)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CommentaryFeatureOverride(APIView):
+    permission_classes = []
+    authentication_classes = [SingleTokenAuthentication]
+    renderer_classes = [JSONRenderer]
+
+    def get(self, request, pk):
+        if not pk:
+            return Response(
+                {"error": "Commentary Id missing in request url."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        commentary_instance = Commentary.objects.filter(pk=pk).first()
+        if not commentary_instance:
+            return Response(
+                {"error": "Commentary does not exsist"}, status=status.HTTP_404_NOT_FOUND
+            )
+        serializer = CommentaryFeatureOverrideSerializer(commentary_instance)
+
+        return Response(serializer.data.get("feature_override"))
+
+    def put(self, request, pk):
+        if not pk:
+            return Response(
+                {"error": "Commentary Id missing in request url."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        commentary_instance = Commentary.objects.filter(pk=pk).first()
+        if not commentary_instance:
+            return Response(
+                {"error": "Commentary does not exsist"}, status=status.HTTP_404_NOT_FOUND
+            )
+
+        commentary_instance = Commentary.objects.filter(pk=pk).first()
```

It's really not great to use `.filter().first()` here. .filter returns a list whereas there should either only be one result or the API should throw an error.

---

As mentioned in the previous comment, it might be better to use the brand id

---

updated